src/test/java/edu/kis/powp/jobs2d/TestJobs2dApp.java

```diff
@@ -49,66 +51,51 @@ private static void setupPresetTests(Application application) {
-        DriverFeature.addDriver("Line Simulator", driver);
-        DriverFeature.getDriverManager().setCurrentDriver(driver);
+        // Transforming Driver for Line Simulator
+        TransformingDriver lineDriver = new TransformingDriver(basicLineDriver);
+
+        // Transforming Driver for Special Line Simulator
+        TransformingDriver specialLineDriver = new TransformingDriver(basicSpecialLineSimulator);
 
-        // Transformed Driver for Line Simulator - scaled to 25% and rotated by 70deg
-        TransformingDriver ScaledAndRotatedDriver = new TransformingDriver(driver);
-        ScaledAndRotatedDriver.addModifier(new ScalingModifier(0.25, 0.25));
-        ScaledAndRotatedDriver.addModifier(new RotationModifier(70));
-        DriverFeature.addDriver("Line Simulator  - scale: 1/4, rotation: 70deg", ScaledAndRotatedDriver);
+        // Driver for Line Simulator
+        DriverFeature.addDriver("Line Simulator", lineDriver);
+        DriverFeature.getDriverManager().setCurrentDriver(lineDriver);
 
         // Driver for Special Line Simulator
-        driver = new LineDriverAdapter(drawerController, LineFactory.getSpecialLine(), "special");
-        DriverFeature.addDriver("Special line Simulator", driver);
+        DriverFeature.addDriver("Special Line Simulator", specialLineDriver);
 
-        // Transformed Driver for Special Line Simulator - flipped in both planes and shifted axes: X(50) Y(-50)
-        TransformingDriver FlippedAndShiftedDriver = new TransformingDriver(driver);
-        FlippedAndShiftedDriver.addModifier(ScalingModifierFactory.createHorizontalFlipModifier());
-        FlippedAndShiftedDriver.addModifier(ScalingModifierFactory.createVerticalFlipModifier());
-        FlippedAndShiftedDriver.addModifier(new ShiftAxesModifier(50, -50));
-        DriverFeature.addDriver("Special Line Simulator  - flipped X and Y, shifted X and Y", FlippedAndShiftedDriver);
+        // Add modifiers to the extensions
+        Extensions.addMenuElementWithCheckbox(app, "Scale: 1/4", new ModifierListener(new ScalingModifier(0.25, 0.25), lineDriver, specialLineDriver));
+        Extensions.addMenuElementWithCheckbox(app, "Rotation: 70deg", new ModifierListener(new RotationModifier(70), lineDriver, specialLineDriver));
+        Extensions.addMenuElementWithCheckbox(app, "Horizontal flip", new ModifierListener(ScalingModifierFactory.createHorizontalFlipModifier(), lineDriver, specialLineDriver));
+        Extensions.addMenuElementWithCheckbox(app, "Vertical flip", new ModifierListener(ScalingModifierFactory.createVerticalFlipModifier(), lineDriver, specialLineDriver));
+        Extensions.addMenuElementWithCheckbox(app, "Shifted X and Y", new ModifierListener(new ShiftAxesModifier(50, -50), lineDriver, specialLineDriver));
+        Extensions.addMenuElementWithCheckbox(app, "Macro", new MacroListener(DriverFeature.getDriverManager()));
```

Comment:
Code duplication:
Extensions.addMenuElementWithCheckbox(app, ..., new ModifierListener(new ..., lineDriver, specialLineDriver));

Try to come up, with a better Extensions class API.

Reply:
How was it resolved?
You could have tried:
Extensions.addTransformationModifier(app, "Scale: 1/4", new ModifierListener(new ScalingModifier(0.25, 0.25))
....
and 
Extensions.addTransformableDriver(lineDriver)
Extensions.addTransformableDriver(specialLineDriver)